CONTRIBUTING.md

```diff
@@ -0,0 +1,368 @@
+# Contributing Guidelines
+
+Thank you for your interest in contributing to the Android distribution
+of the AWS SDK. Whether it's a bug report, new feature,
+correction, or additional documentation, the project maintainers at AWS
+greatly value your feedback and contributions.
+
+Please read through this document before submitting any issues or pull
+requests. Doing so will help to ensure that the project maintainers have
+all information necessary to effectively respond to your bug report or
+contribution.
+
+* [Getting Started](#getting-started)
+* [Tools](#tools)
+* [SDK Fundamentals](#sdk-fundamentals)
+* [Workflows](#workflows)
+* [Reporting Bugs/Feature Requests](#reporting-bugs-feature-requests)
+* [Contributing via Pull Requests](#contributing-via-pull-requests)
+
+## Getting Started
+
+First, ensure that you have installed the latest stable version of Android
+Studio / the Android SDK.
+
+Configure your environment, so that the `ANDROID_HOME` and `JAVA_HOME`
+environment variables are set. A convenient way of doing this is to add them
+into `~/.bashrc`. On a Mac, the SDK and Java installation used by the SDK may
+be found:
+
+```
+export ANDROID_HOME=~/Library/Android/sdk
+export JAVA_HOME=/Applications/Android\ Studio.app/Contents/jre/jdk/Contents/Home
+```
+Note: JDK 11, 12, 13, etc. have known issues and are not supported.
+
+Now, clone the AWS SDK for Android project from GitHub.
+
+```
+git clone git@github.com:aws-amplify/aws-sdk-android.git
+```
+Load this project into Android Studio by selecting File > Open, and choosing
+the root directory of the project (`aws-sdk-android`). Alternately, cd into this
+top-level directory.
+
+In Android Studio, build the project by clicking the Hammer icon, "Make Project
+⌘F9". If working on the command line, you can do the same thing via:
+
+```
+./gradlew build
+```
+
+## Tools
+[Gradle](https://gradle.org) is used for all [build and dependency management](https://developer.android.com/studio/build).
+
+Some widely used dependencies are:
+
+1. [org.json](https://developer.android.com/reference/org/json/JSONObject) is
+   baked into Android, and is used for all modeling with JSON.
+2. [Gson](https://github.com/google/gson/blob/master/README.md#gson) is used
+   for serialization and deserialization.
+3. [OkHttp](https://github.com/square/okhttp#okhttp) is used for network operations.
```

Comment:
These are widely used in Amplify, but, in the SDK:
- Gson is used only once (the Android SDK prefers Jackson, since it was originally forked from the Java SDK.)
- OkHttp isn't used at all, but rather Apache HTTP components are preferred.

I might suggest that we just remove this entire "some widely used dependencies ...." section.

Reply:
done